src/lib/encoder.ts

```diff
@@ -52,10 +52,13 @@ import {
   UNKNOWN_VERIFICATION_METHOD,
 } from '../constants/constants';
 import { getVerificationMethod, hashData, countNumberOfBytes } from './utils';
+import { ERC725JSONSchemaValueType } from '../types/ERC725JSONSchema';
 
 const abiCoder = AbiCoder;
 
-const bytesNRegex = /Bytes(\d+)/;
+const uintNValueTypeRegex = /^uint(?:8|[1-9]\d|1[0-9]{2}|2[0-4]\d|25[0-6])$/;
```

Comment:
I have doubts over this one

- it is hard to read
- it catches values which should not be ok (uint23...)

I suggest to use just a catch to get N and then test N with proper maths